app/src/main/java/ch/epfl/sdp/drone3d/ui/map/offline/ManageOfflineMapActivity.kt

```diff
@@ -0,0 +1,147 @@
+package ch.epfl.sdp.drone3d.ui.map.offline
+
+import android.os.Bundle
+import android.widget.ProgressBar
+import android.widget.TextView
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import ch.epfl.sdp.drone3d.R
+import ch.epfl.sdp.drone3d.map.offline.OfflineMapSaver
+import ch.epfl.sdp.drone3d.map.offline.OfflineMapSaverImpl
+import ch.epfl.sdp.drone3d.ui.ToastHandler
+import ch.epfl.sdp.drone3d.ui.map.BaseMapActivity
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import com.mapbox.mapboxsdk.Mapbox
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.offline.OfflineRegion
+import com.mapbox.mapboxsdk.offline.OfflineRegionError
+import com.mapbox.mapboxsdk.offline.OfflineRegionStatus
+import timber.log.Timber
+import java.lang.StringBuilder
+import java.lang.System.currentTimeMillis
+import kotlin.math.min
+
+class ManageOfflineMapActivity : BaseMapActivity(), OnMapReadyCallback {
+
+    companion object{
+        private const val DOWNLOAD_STATUS_TIME_DELAY = 1000
+    }
+
+    private lateinit var offlineMapSaver: OfflineMapSaver
+    private lateinit var mapboxMap: MapboxMap
+    private lateinit var downloadButton: FloatingActionButton
+    private var timeOfLastDownloadToast = 0L
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        Mapbox.getInstance(this, getString(R.string.mapbox_access_token))
+        super.initMapView(savedInstanceState, R.layout.activity_manage_offline_map, R.id.mapView)
+
+        mapView.contentDescription = getString(R.string.map_not_ready)
+        mapView.getMapAsync(this)
+
+        downloadButton = findViewById(R.id.buttonToSaveOfflineMap)
+        //We must wait for the map to be available to download
+        downloadButton.isEnabled = false
+
+
+        //Create a "back button" in the action bar up
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
```

Comment:
I don't think this is needed, in other activities the button is there without this

Reply:
I think that you are right, thanks